src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationFix.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.PXFieldAttributes;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Acuminator.Analyzers.StaticAnalysis.DacKeyFieldDeclaration
+{
+	[Shared]
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class KeyFieldDeclarationFix : CodeFixProvider
+	{
+		private const string IsKey = nameof(PX.Data.PXDBFieldAttribute.IsKey);
+		
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1055_DacKeyFieldBound.Id);
+
+		public override  Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			return Task.Run(() =>
+			{
+				var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1055_DacKeyFieldBound.Id);
```

Comment:
Your code fix is registered for `PX1055` only, so all these checks are not needed. You always get a PX1055 diagnostic here, otherwise code fix won't be registered at all.

Reply:
@dropsonic  I suppose that's not the case here - @saloponov wants to find the diagnostic  from the context because he passed with it additional locations to code fix

Reply:
@dropsonic I use `diagnostic.AdditionalLocations`.